nestkernel/conn_builder.cpp

```diff
@@ -382,9 +382,18 @@ nest::ConnBuilder::change_connected_synaptic_elements( index sgid,
 void
 nest::ConnBuilder::connect()
 {
+  if ( kernel().model_manager.connector_requires_symmetric( synapse_model_ )
+    && not symmetric_ && not is_symmetric() )
```

Comment:
@heplesser Good points! I think now we found all conditions that need to be checked in order to ensure that an all-to-all connection is symmetric. I would also vote for a solution where we prohibit both array and random parameters, rather than allowing those in general and check them for symmetry case by case.

So I would imagine our solution like this: Change the `is_symmetric()` function of the all-to-all ConnBuilder to
```
bool
is_symmetric() const
{
  if( sources != targets )
    return false;

  if( parameters are arrays or randomized )
    return false;
 
  return true;
}
```
and adjust the `BadProperty`-Message to `"This synapse model requires symmetric (or suitable uniform all-to-all) connections"`

The check in `is_symmetric()` will only be performed, if the first two conditions of the if-statement are met, ergo only if a synapse model requires symmetric connections and the `symmetric` flag is set to `false`

Would you agree on this general plan? In this case we would need to figure out how to implement the two `if` statements. @jougs could help with this first one, any idea how to check for the second one most efficiently?